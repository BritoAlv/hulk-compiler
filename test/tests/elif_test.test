print(tan(PI) ** 2 + cot(PI) ** 2); #error
let msg = "Hello World" in print(msg);
let number = 42, text = "The meaning of life is" in
    print(text @ number);
let number = 42 in
    let text = "The meaning of life is" in
        print(text @ number);
let number = 42 in (
    let text = "The meaning of life is" in (
            print(text @ number);
        )
    );
let a = 6, b = a * 7 in print(b);
let a = 6 in
    let b = a * 7 in
        print(b);
let a = 5, b = 10, c = 20 in {
    print(a+b);
    print(b*c);
    print(c/a);
};
let a = (let b = 6 in b * 7) in print(a);
print(let b = 6 in b * 7);
let a = 20 in {
    let a = 42 in print(a);
    print(a);
};
let a = 7, a = 7 * 6 in print(a);
let a = 7 in
    let a = 7 * 6 in
        print(a);
let a = 0 in {
    print(a);
    a := 1;
    print(a);
};
let a = 0 in
    let b = a := 1 in {
        print(a);
        print(b);
    };
let a = 42 in if (a % 2 == 0) print("Even") else print("odd");
let a = 42 in print(if (a % 2 == 0) "even" else "odd");
let a = 42 in
    if (a % 2 == 0) {
        print(a);
        print("Even");
    }
    else print("Odd");
let a = 42, mod = a % 3 in # error
    print(
        if (mod == 0) "Magic"
        elif (mod % 3 == 1) "Woke"
        else "Dumb"
    );
let a = 10 in while (a >= 0) {
    print(a);
    a := a - 1;
}