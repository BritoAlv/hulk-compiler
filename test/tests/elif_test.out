There was a parsing error :
Near error on line 0, star:
0 print(tan(PI) ** 2 + cot(PI) ** 2); #error
1 let msg = "Hello World" in print(msg);

0 print(tan(PI) ** 2 + cot(PI) ** 2); #error
0                ^
Trying to recovery from error
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert false
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert number
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert id
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert base
----------------
----------------
Seq: 
insert self
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
delete star *  at line 0 at position 15 
----------------
----------------
Seq: 
insert true
----------------
----------------
Seq: 
insert string
----------------
----------------
Seq: 
insert string
----------------
